src/main/java/com/example/database/services/CardService.java

```diff
@@ -62,6 +104,27 @@ public AddressModel getAddressModel(Long id) {
                 .convertToModel(address);
     }
 
+    private Uni<List<Long>> getFilteredResults(SearchRequest request) {
+        return searchService
+                .searchItemsBySearchRequest(request).onItem().apply(json -> {
+
+                    //null check on json
+                    if (json == null || json.isEmpty() || json.getJsonObject("hits") == null

```

Comment:
How about do it inline, null safe, chaining way?
```suggestion
                    return Optional.ofNullable(json)
                            .map(j -> j.getJsonObject("hits"))
                            .map(hits -> hits.getJsonArray("hits"))
                            .orElse(new JsonArray())
                            .stream()
                            .filter(o -> o instanceof JsonObject)
                            .map(JsonObject.class::cast)
                            .map(o -> parseLong(o.getString("_id")))
                            .collect(toList());

```

Reply:
Changed! I must learn this optional -> map -> orElse construct